onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -72,35 +74,21 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
     std::string ov_string = performance_test_config.run_config.ep_runtime_config_string;
 #endif  // defined(_MSC_VER)
     int num_threads = 0;
-    std::istringstream ss(ov_string);
-    std::string token;
-    while (ss >> token) {
-      if (token == "") {
-        continue;
-      }
-      auto pos = token.find("|");
-      if (pos == std::string::npos || pos == 0 || pos == token.length()) {
-        ORT_THROW(
-            "[ERROR] [OneDNN] Use a '|' to separate the key and value for the "
-            "run-time option you are trying to use.\n");
-      }
-
-      auto key = token.substr(0, pos);
-      auto value = token.substr(pos + 1);
-
-      if (key == "num_of_threads") {
-        std::stringstream sstream(value);
+    if (!ParseSessionConfigs(ov_string, provider_options, {"num_of_threads"})) {
+      ORT_THROW(
+          "[ERROR] Use a '|' to separate the key and value for the "
+          "run-time option you are trying to use.\n");
```

Comment:
ParseSessionConfigs now throws and could return void as there's no path returning `false` anymore. 

given that, we shouldn't need any throws about the formatting in this file anymore.

Reply:
make sense.